spacewalk/setup/spacewalk-setup.spec

```diff
@@ -244,9 +244,13 @@ if grep 'authn_spacewalk' /etc/cobbler/modules.conf > /dev/null 2>&1; then
     sed -i 's/module = authn_spacewalk/module = authentication.spacewalk/' /etc/cobbler/modules.conf
 fi
 
-# if old /etc/cobbler/settings exists, we need to perform migration
-if [ -e /etc/cobbler/settings ]; then
-    echo "* Creating a backup of /etc/cobbler/settings file to /etc/cobbler/settings.before-migration-backup before migrating settings"
+# When upgrading to Cobbler 3.3.3, the old /etc/cobbler/settings config file from previous Cobbler version
+# is removed as it not existing anymore in the new version, but a copy is kept with the local changes done
+# at /etc/cobbler/settings.rpmsave. If this file exists, it means we need to perform the migration of these
+# settings and also trigger the migration of stored Cobbler collections.
+if [ ! -f /etc/cobbler/settings -a -f /etc/cobbler/settings.rpmsave ]; then
```

Comment:
Since we are going to release this as a patch on top of 2022.10, are you sure `/etc/cobbler/settings` will never exist?

Reply:
As soon as Cobbler 3.3.3 is installed, the file should be gone.

If `/etc/cobbler/settings` exists prior to Cobbler 3.3.3 upgrade, then it will be renamed to `settings.rpmsave` during the Cobbler upgrade before reaching this point, as we are making this package to `Requires(post)`.

If Cobbler 3.3.3 is already installed, so current users with Uyuni 2022.10 already installed, the `/etc/cobbler/settings` file should not be existing and already renamed to `settings.rpmsave` during the upgrade to Uyuni 2022.10.

Reply:
ACK, looks good then.